### PR TITLE
METRON-1427: Add support for storm 1.1 and hdp 2.6

### PR DESCRIPTION
**Pull request by cestella**
## Contributor Comments
Right now our ambari mpack won't run cleanly on HDP 2.6 and Storm 1.1 because of some classpath issues and not supporting the stack.  We should migrate fulldev to run 2.6 (while still working with 2.5) and validate that we work with storm 1.1.   The only code changes in here is relocating and shading jackson (kerberized storm 1.1 requires a conflicting version of jackson from ES and us, so we shade and relocate it in parsers, but didn't in profiler and elasticsearch uber jars.  This PR corrects that).  

You can validate this by:
* Running up full-dev and ensuring data flows through
* Kerberizing full-dev and ensure things still work

I have done the above, but you can feel free to do the same.  
## Pull Request Checklist

Thank you for submitting a contribution to Apache Metron.  
Please refer to our [Development Guidelines](https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=61332235) for the complete guide to follow for contributions.  
Please refer also to our [Build Verification Guidelines](https://cwiki.apache.org/confluence/display/METRON/Verifying+Builds?show-miniview) for complete smoke testing guides.  

In order to streamline the review of the contribution we ask you follow these guidelines and ask you to double check the following:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [x] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [x] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

#### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.
It is also recommended that [travis-ci](https://travis-ci.org) is set up for your personal repository such that your branches are built there before submitting a pull request.

### CODE REVIEW

File: metron-deployment/roles/ambari_config/vars/single_node_vm.yml [hunk 1]
```diff
@@ -87,6 +87,11 @@ configurations:
       supervisor.slots.ports: "[6700, 6701, 6702, 6703, 6704, 6705]"
       storm.local.dir: '{{ storm_local_dir }}'
       topology.classpath: '{{ topology_classpath }}'
+      storm.cluster.metrics.consumer.register: '[{"class": "org.apache.storm.metric.LoggingMetricsConsumer"}]'
```
Comment (justinleet) [inline metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: Could you add a comment or explanation about why the changes to the Storm config?  The other changes are pretty straightforward, but it's not obvious why we need config changes (unless you already know its to deal with Storm issues around metrics and killing JMX from the childopts).

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/ELASTICSEARCH/5.6.2/package/scripts/service_check.py [hunk 2]
```diff
@@ -32,7 +32,7 @@ def service_check(self, env):
         doc = '{"name": "Ambari Smoke test"}'
         index = "ambari_smoke_test"
 
-        Logger.info("Running Elastic search service check", file=sys.stdout)
+        Logger.info("Running Elastic search service check")
```
Comment (anandsubbu) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/ELASTICSEARCH/5.6.2/package/scripts/service_check.py]: Hi @cestella , would it be a good idea to leave this unchanged since @mmiklavc has PR #904 which addresses this issue?

Comment (cestella) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/ELASTICSEARCH/5.6.2/package/scripts/service_check.py]: It would be normally, but I couldn't test it in kerberos without it (status check fails when kerberizing because of this).  As soon as master is fixed and #904 is in (has my +1 already), I'll merge.

File: metron-deployment/roles/ambari_config/vars/single_node_vm.yml (hunk 1 above)
Comment (cestella) [reply metron-deployment/roles/ambari_config/vars/single_node_vm.yml]: Sounds good, did that

**Comment by mraliagha** [comment]
@cestella  Would it be possible to target Storm 1.2 (which was released 2 days ago) as well?

**Comment by mmiklavc** [comment]
@Ali - I think the reason for Storm 1.1 is that it's the version that comes
with HDP 2.6.
https://docs.hortonworks.com/HDPDocuments/HDP2/HDP-2.6.4/bk_release-notes/content/comp_versions.html

**Comment by anandsubbu** [comment]
Hi @cestella , I did a 12-node deploy on CentOS 7 with this patch. Post-kerberization, I noticed the following errors in Metron REST. Is this a related issue or a different one?

```
18/01/25 10:16:02 ERROR boot.SpringApplication: Application startup failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'alertController': Unsatisfied dependency expressed through field 'alertService'; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'alertServiceImpl' defined in URL [jar:file:/usr/metron/0.4.3/lib/metron-rest-0.4.3.jar!/org/apache/metron/rest/service/impl/AlertServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kafkaServiceImpl' defined in URL [jar:file:/usr/metron/0.4.3/lib/metron-rest-0.4.3.jar!/org/apache/metron/rest/service/impl/KafkaServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaProducer' defined in class path resource [org/apache/metron/rest/config/KafkaConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.apache.kafka.clients.producer.KafkaProducer]: Factory method 'kafkaProducer' threw exception; nested exception is org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:569)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:349)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1219)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:543)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:197)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:751)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:861)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:122)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:761)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:371)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1186)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1175)
	at org.apache.metron.rest.MetronRestApplication.main(MetronRestApplication.java:29)
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'alertServiceImpl' defined in URL [jar:file:/usr/metron/0.4.3/lib/metron-rest-0.4.3.jar!/org/apache/metron/rest/service/impl/AlertServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kafkaServiceImpl' defined in URL [jar:file:/usr/metron/0.4.3/lib/metron-rest-0.4.3.jar!/org/apache/metron/rest/service/impl/KafkaServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaProducer' defined in class path resource [org/apache/metron/rest/config/KafkaConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.apache.kafka.clients.producer.KafkaProducer]: Factory method 'kafkaProducer' threw exception; nested exception is org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:749)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:189)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1148)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1051)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1128)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1056)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:566)
	... 19 more
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'kafkaServiceImpl' defined in URL [jar:file:/usr/metron/0.4.3/lib/metron-rest-0.4.3.jar!/org/apache/metron/rest/service/impl/KafkaServiceImpl.class]: Unsatisfied dependency expressed through constructor parameter 2; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaProducer' defined in class path resource [org/apache/metron/rest/config/KafkaConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.apache.kafka.clients.producer.KafkaProducer]: Factory method 'kafkaProducer' threw exception; nested exception is org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:749)
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:189)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1148)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1051)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1128)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1056)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:835)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:741)
	... 32 more
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'kafkaProducer' defined in class path resource [org/apache/metron/rest/config/KafkaConfig.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.apache.kafka.clients.producer.KafkaProducer]: Factory method 'kafkaProducer' threw exception; nested exception is org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:599)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1128)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1023)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:510)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:482)
	at org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:306)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:230)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:302)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:207)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1128)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1056)
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:835)
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:741)
	... 46 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.apache.kafka.clients.producer.KafkaProducer]: Factory method 'kafkaProducer' threw exception; nested exception is org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:189)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:588)
	... 59 more
Caused by: org.apache.kafka.common.KafkaException: Failed to construct kafka producer
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:335)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:163)
	at org.apache.metron.rest.config.KafkaConfig.kafkaProducer(KafkaConfig.java:119)
	at org.apache.metron.rest.config.KafkaConfig$$EnhancerBySpringCGLIB$$7491327d.CGLIB$kafkaProducer$4(<generated>)
	at org.apache.metron.rest.config.KafkaConfig$$EnhancerBySpringCGLIB$$7491327d$$FastClassBySpringCGLIB$$d2cc2647.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:228)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:356)
	at org.apache.metron.rest.config.KafkaConfig$$EnhancerBySpringCGLIB$$7491327d.kafkaProducer(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:162)
	... 60 more
Caused by: java.lang.IllegalArgumentException: No enum constant org.apache.kafka.common.protocol.SecurityProtocol.PLAINTEXTSASL
	at java.lang.Enum.valueOf(Enum.java:238)
	at org.apache.kafka.common.protocol.SecurityProtocol.valueOf(SecurityProtocol.java:28)
	at org.apache.kafka.common.protocol.SecurityProtocol.forName(SecurityProtocol.java:89)
	at org.apache.kafka.clients.ClientUtils.createChannelBuilder(ClientUtils.java:79)
	at org.apache.kafka.clients.producer.KafkaProducer.<init>(KafkaProducer.java:277)
	... 72 more
```

**Comment by cestella** [comment]
@anandsubbu Yeah, That's that lingering craziness where sometimes Kafka is expecting `PLAINTEXTSASL` and sometimes `SASL_PLAINTEXT`.  I just went ahead and normalized it to whatever the kafka library enum had and translated accordingly.  Try it out now after the most recent commit (also, I merged in master, so Mike's changes in #904 are in).

**Comment by cestella** [comment]
ok, I ran this guy up in kerberos and regular and tooled around a bit; ensured the alerts UI worked in both.

**Comment by mmiklavc** [comment]
I'm testing now as well for good measure since this is our main dev testing environment. We should probably take a little extra care with the breadth of changes in this and https://github.com/apache/metron/pull/903

**Comment by nickwallen** [comment]
Running this up on Ubuntu, I ran into two issues.

(1) JDK Install Failed on Ubuntu - I do not think this was caused by this PR.  This is something that I should have hit in #903, but the changes there never brought this bug to light.  Fortunately, it is an easy fix and I will open it as a separate PR.

(2) After the Ambari deployment completes, it begins to install the sensors.  It queries Ambari for information to configure the sensors and in one of those queries it hits this problem.  I am not sure exactly what the problem is yet.
```
TASK [ambari_gather_facts : Ask Ambari: hdfs_url] ******************************
ok: [node1]

TASK [ambari_gather_facts : set_fact] ******************************************
fatal: [node1]: FAILED! =>{  
   "failed":true,
   "msg":"the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: {u'status': 200, u'content_type': u'text/plain', u'set_cookie': u'AMBARISESSIONID=5tlv0y7btgc24krm5ugdpdgj;Path=/;HttpOnly', u'expires': u'Thu, 01 Jan 1970 00:00:00 GMT', u'vary': u'Accept-Encoding, User-Agent', u'user': u'admin', u'pragma': u'no-cache', u'x_frame_options': u'DENY', u'x_xss_protection': u'1; mode=block', u'url': u'http://node1:8080/api/v1/clusters/metron_cluster/configurations?type=core-site&tag=TOPOLOGY_RESOLVED', u'changed': False, u'x_content_type_options': u'nosniff', u'content': u'{\\n  \"href\" : \"http://node1:8080/api/v1/clusters/metron_cluster/configurations?type=core-site&tag=TOPOLOGY_RESOLVED\",\\n  \"items\" : [\\n    {\\n      \"href\" : \"http://node1:8080/api/v1/clusters/metron_cluster/configurations?type=core-site&tag=TOPOLOGY_RESOLVED\",\\n      \"tag\" : \"TOPOLOGY_RESOLVED\",\\n      \"type\" : \"core-site\",\\n      \"version\" : 2,\\n      \"Config\" : {\\n        \"cluster_name\" : \"metron_cluster\",\\n        \"stack_id\" : \"HDP-2.6\"\\n      },\\n      \"properties\" : {\\n        \"fs.defaultFS\" : \"hdfs://node1:8020\",\\n        \"fs.trash.interval\" : \"360\",\\n        \"ha.failover-controller.active-standby-elector.zk.op.retries\" : \"120\",\\n        \"hadoop.custom-extensions.root\" : \"/hdp/ext/{{major_stack_version}}/hadoop\",\\n        \"hadoop.http.authentication.simple.anonymous.allowed\" : \"true\",\\n        \"hadoop.proxyuser.hbase.groups\" : \"*\",\\n        \"hadoop.proxyuser.hbase.hosts\" : \"*\",\\n        \"hadoop.security.auth_to_local\" : \"DEFAULT\",\\n        \"hadoop.security.authentication\" : \"simple\",\\n        \"hadoop.security.authorization\" : \"false\",\\n        \"hadoop.security.key.provider.path\" : \"\",\\n        \"io.compression.codecs\" : \"org.apache.hadoop.io.compress.GzipCodec,org.apache.hadoop.io.compress.DefaultCodec,org.apache.hadoop.io.compress.SnappyCodec\",\\n        \"io.file.buffer.size\" : \"131072\",\\n        \"io.serializations\" : \"org.apache.hadoop.io.serializer.WritableSerialization\",\\n        \"ipc.client.connect.max.retries\" : \"50\",\\n        \"ipc.client.connection.maxidletime\" : \"30000\",\\n        \"ipc.client.idlethreshold\" : \"8000\",\\n        \"ipc.server.tcpnodelay\" : \"true\",\\n        \"mapreduce.jobtracker.webinterface.trusted\" : \"false\",\\n        \"net.topology.script.file.name\" : \"/etc/hadoop/conf/topology_script.py\"\\n      },\\n      \"properties_attributes\" : {\\n        \"final\" : {\\n          \"fs.defaultFS\" : \"true\"\\n        }\\n      }\\n    }\\n  ]\\n}', u'connection': u'close', u'msg': u'OK (unknown bytes)', u'redirected': False, u'cache_control': u'no-store'}: 'major_stack_version' is undefined\n\nThe error appears to have been in '/Users/nallen/tmp/metron-pr907/metron-deployment/ansible/roles/ambari_gather_facts/tasks/main.yml': line 82, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- set_fact:\n  ^ here\n"
}
```

**Comment by nickwallen** [comment]
For (2), the problem is that the HDFS configuration in Ambari has embedded 'mustache handlebars' (see `{{major_stack_version}}`) that when returned from an API call confuses Ansible.  We need some way to strip that out or ignore it.

screen shot 2018-01-29 at 2 31 44 pm

**Comment by cestella** [comment]
For 2, that's the same error @mmiklavc hit on centos.  It's very interesting that I'm not hitting it.

**Comment by nickwallen** [comment]
FYI @cestella I submitted a fix against your PR branch that should address the issue with the embedded handlebars in the Ambari response.

**Comment by anandsubbu** [comment]
Hi @cestella , with the latest commit having the normalized producer, I was able to spin up a 12-node Centos 7 cluster, install with HDP 2.6 base and kerberize as well without issues. Ran the standard squid sensor test and was able to see events in the Alerts UI.

+1 from my end.

**Comment by cestella** [comment]
@nickwallen Awesome, your PR is merged in here and I'll spin it up one last time to verify that it works for 2.0.0.2.

@anandsubbu Awesome, thanks!

**Comment by nickwallen** [comment]
I also just merged #913 so I will run this up just to be sure the whole package is working in the Ubuntu dev environment.

**Comment by nickwallen** [comment]
+1 Ran it up with all of our fixes in the Ubuntu dev environment.  Works great.  Thanks!

**Comment by mmiklavc** [comment]
+1
I ran it up on 2 separate Macs, one running Ansible 2.2.2.0 and the other 2.0.0.2. Both spun up without issue today.
